src/uct/cuda/cuda_ipc/cuda_ipc_cache.c

```diff
@@ -183,6 +363,32 @@ uct_cuda_ipc_get_remote_cache(pid_t pid, uct_cuda_ipc_cache_t **cache)
     return status;
 }
 
+ucs_status_t uct_cuda_ipc_close_memhandle(uct_cuda_ipc_cache_region_t *region)
+{
+    ucs_status_t status;
+#if HAVE_CUDA_FABRIC
+    if (region->key.ph.handle_type == UCT_CUDA_IPC_KEY_HANDLE_TYPE_VMM) {
+        status = UCT_CUDADRV_FUNC_LOG_ERR(cuMemUnmap(
+                    (CUdeviceptr)region->mapped_addr, region->key.b_len));
+        status = UCT_CUDADRV_FUNC_LOG_ERR(cuMemAddressFree(
+                    (CUdeviceptr)region->mapped_addr, region->key.b_len));
+    } else if (region->key.ph.handle_type == UCT_CUDA_IPC_KEY_HANDLE_TYPE_LEGACY) {
+        status = UCT_CUDADRV_FUNC_LOG_ERR(cuIpcCloseMemHandle(
+                    (CUdeviceptr)region->mapped_addr));
+    } else {
+        /* Ideally we call cuMemFreeAsync on imported pointer region here but
+         * handles can be closed after device context has been destroyed which
+         * would cause cuMemFreeAsync to fail for a given stream */
+        status = UCS_OK;
+    }
+#else
+     status = UCT_CUDADRV_FUNC_LOG_ERR(cuIpcCloseMemHandle(
+                 (CUdeviceptr)region->mapped_addr));
+#endif
+
+     return status;
+}
```

Comment:
```suggestion
ucs_status_t uct_cuda_ipc_close_memhandle(uct_cuda_ipc_cache_region_t *region)
{
    ucs_status_t status;
#if HAVE_CUDA_FABRIC
    if (region->key.ph.handle_type == UCT_CUDA_IPC_KEY_HANDLE_TYPE_VMM) {
        status = UCT_CUDADRV_FUNC_LOG_ERR(cuMemUnmap(
                    (CUdeviceptr)region->mapped_addr, region->key.b_len));
        status = UCT_CUDADRV_FUNC_LOG_ERR(cuMemAddressFree(
                    (CUdeviceptr)region->mapped_addr, region->key.b_len));
    } else if (region->key.ph.handle_type != UCT_CUDA_IPC_KEY_HANDLE_TYPE_LEGACY) {
        /* Ideally we call cuMemFreeAsync on imported pointer region here but
         * handles can be closed after device context has been destroyed which
         * would cause cuMemFreeAsync to fail for a given stream */
        status = UCS_OK;
    }
#endif
     status = UCT_CUDADRV_FUNC_LOG_ERR(cuIpcCloseMemHandle(
                 (CUdeviceptr)region->mapped_addr));

     return status;
}
```